### ISSUE TITLE
LOSS can not convergence

### ISSUE THREAD

**Issue by Haosouth** [xchhuang/pytorch_sliced_wasserstein_loss#1]
Hello, thank you very much for writing sliced_wasserstein_loss as a pytorch version. I tried to migrate it to my style transfer network and replace the original Gram matrix used to describe style features. Although the code runs successfully, Loss does  I have not been able to converge. I have made some changes to the code. I don’t know if my understanding of the code is wrong. I hope you can help me. Thank you very much. The following are my changes:
```
 def forward(self, input, target):
        self.update_slices(input)
        loss = 0.0
        target_ = self.compute_target(target)
        for idx, l in enumerate(input):
            cur_l = self.compute_proj(l, idx, 1)
            loss += F.mse_loss(cur_l, target_[idx])
        return loss
```
Call this LOSS：
```
slicing_torch = Slicing_torch(device, repeat_rate=1)
loss_style = slicing_torch(Ics_feats[0], style_feats[0])
```

**Comment by xchhuang** [comment]
I see. In my case, running the texture synthesis on `data/SlicedW/input.jpg` in the repo would give me the following loss curve. And the final synthesis result is shown in 1st row of `Sample Results` section in readme.

<img src="https://user-images.githubusercontent.com/22114321/139524612-42ae0dbe-d33f-4137-9e66-e1994e0571c7.jpg" width="400" height="300" />

Would you share your loss curve, final synthesis result on the same data?

**Comment by Haosouth** [comment]
![78`_JI%$39~$XA$O~B%$E{2](https://user-images.githubusercontent.com/56715037/141415015-b1391f0b-5484-4c87-9c2b-1438d6df7949.png)
I am sorry to reply to you for so long and this is my loss curve,

**Comment by Haosouth** [comment]
![image](https://user-images.githubusercontent.com/56715037/141415401-bbb58045-e568-4fc8-86b5-185fa6fbe2a3.png)
Maybe is the number of times my training is not enough?

**Comment by xchhuang** [comment]
No problem, yes the loss should not be that large. 

From my understanding, you don't need to call `self.update_slices() and self.compute_target()` every time in `forward`. Instead, you will need that after each iteration of LBFGS, that's why I move them outside `forward`.

The issue may come from your new `Slicing_torch` class, but I am not sure without knowing more details.
